tchannel/tornado/dispatch.py

```diff
@@ -82,21 +85,28 @@ def handle_call(self, request, connection):
             request,
         )
 
-        endpoint = self.endpoints[request.endpoint]
-
+        handler = self.handlers[request.endpoint]
+        if not request.headers.get('as', None) == handler.req_serializer.name:
+            raise gen.Return(connection.send_error(
+                ErrorCode.bad_request,
+                "Invalid arg scheme in request header",
+                request.id,
```

Comment:
And then you need a separate `return` statement.

(You can do `return` without any arguments inside a generator.)